writer.go

```diff
@@ -27,6 +28,7 @@ type Writer struct {
 
 // WriterConfig is a configuration type used to create new instances of Writer.
 type WriterConfig struct {
+	newPartitionWriter func(partition int, config WriterConfig) partitionWriter
```

Comment:
It's considered good practice in Go to put unexported symbols after exported ones, in order to first present to the reader the API that it can use.

Reply:
👍 I'll open a new PR to fix these.